### DISCUSSION TITLE
Crazyradio to Cfclient Connetion Problem

### DISCUSSION THREAD

**Discussion by WoodyloveBo** [bitcraze/discussions#1051] [Q&A]
Hi. I am using linux on wsl. 

I installed CFclient, plugged in the crazyradio dongle, and tried to scan, but the following message appears and scanning does not work. I would appreciate it if you could let me know what the problem is
![1111](https://github.com/bitcraze/discussions/assets/143582301/ad2a89da-d58e-40e2-b41a-a6a7452e41e3)

**Comment by WoodyloveBo** [comment]
In the process of setting USB permissions, the /etc/udev/rules.d/99-bitcraze.rules file was created, but when I reload-rules, the following error occurred. "Failed to send reload request : No such file or directory"

**Comment by ataffanel** [comment]
It looks like there is no USB in your VM. Have you installed USBIPD to pass the radio to the radio to the wsl VM? It is a required step, it it documented both on [Microsoft website](https://learn.microsoft.com/en-us/windows/wsl/connect-usb) and [USBIPD github](https://github.com/dorssel/usbipd-win/wiki/WSL-support).

To check that it works, you can use lsusb to see if you can find the radio:
```
❯ lsusb -d 1915:7777
Bus 009 Device 034: ID 1915:7777 Nordic Semiconductor ASA Bitcraze Crazyradio (PA) dongle
```

**Reply by Amin-Yazdanshenas** [reply]
I have the same issue the following command gives
```
lsusb -d 1915:7777
Bus 001 Device 004: ID 1915:7777 Nordic Semiconductor ASA Crazyradio 2.0
``` 
but the error persists
 
```
sudo udevadm control --reload-rules
Failed to send reload request: No such file or directory
```
and when I try the cfclient I can't get it to work with crazyradio
```
[Errno 13] Access denied (insufficient permissions)
INFO:cflib.drivers.cfusb:Looking for devices....
INFO:cflib.crtp.prrtdriver:Initialized PRRT driver.
[Errno 13] Access denied (insufficient permissions)
INFO:cflib.drivers.cfusb:Looking for devices....
INFO:cflib.crtp.prrtdriver:Initialized PRRT driver.
```

**Reply by Amin-Yazdanshenas** [reply]
I found the solution.

run the following command and it should work.
```
sudo service udev restart
sudo udevadm control --reload-rules
sudo udevadm trigger
```

**Reply by WoodyloveBo** [reply]
Thank you. I solved the problem by installing ipdusb and sending data.

**Reply by knmcguire** [reply]
Yes indeed, the instructions of usbipd also state that you should do that. Might be handy to have this done every time a new WSL terminal is open... haven't done that myself though.
